sys/kern/vm_map.c

```diff
@@ -87,13 +86,13 @@ vm_map_t *vm_map_new(void) {
   return map;
 }
 
-vm_map_entry_t *vm_map_entry_alloc(vm_object_t *obj, vaddr_t start, vaddr_t end,
+vm_map_entry_t *vm_map_entry_alloc(vm_aref_t aref, vaddr_t start, vaddr_t end,
```

Comment:
Since in majority cases you pass `VM_AREF_EMPTY` to `vm_map_entry_alloc` I'd suggest removing this argument and implement `vm_map_entry_copy` separately.

Reply:
done.